recipe/meta.yaml

```diff
@@ -12,24 +12,27 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 2
+  number: 3
   skip: true  # [win]
   skip: true  # [py2k]
 
 requirements:
   build:
     - {{ compiler('cxx') }}
+    - python                                # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}    # [build_platform != target_platform]
+    - mysql                                 # [build_platform != target_platform]
   host:
     - python
     - wheel
     - setuptools
     - setuptools_scm 6.0.1
     - setuptools_scm_git_archive
     - libtiledb-sql >=0.10
+    - libxcrypt
```

Comment:
I'm also confused by this addition. From reading https://github.com/conda-forge/linux-sysroot-feedstock/issues/52 and linked commits (eg https://github.com/cropsinsilico/yggdrasil/commit/01eda22be0894309c427998e22276a054b52e071), my understanding is that this new requirement for libxcrypt should only affect python builds < 3.8. However, in the builds for the previous commit, the only failure was for Python 3.9 (3.8 and 3.10 both passed). I agree that the error message `cannot find -lcrypt: No such file or directory` implies a missing libxcrypt, but am puzzled why this only affected the python 3.9 build.

Reply:
I  had manually upgraded to cos7 in an effor to resolve the issue. I believe if I drop it from the host the issue will re-appear. Let's see

Reply:
Indeed the libxcrypto issue is back if I drop it from being a requirement

Reply:
I think we can make this `# [linux]`-only?

Related discussion: https://github.com/conda-forge/python-feedstock/pull/676/files#r1491772327

Reply:
The reason that the other linux builds failed in Dimitris earlier runs is that there is an implicit dependency on `libxcrypt` for the non-PyPy builds, so it is available:
```
2024-02-20T10:22:47.7630761Z     libxcrypt:                4.4.36-hd590300_1           conda-forge
```

(via [this build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=880631&view=logs&j=9a864fd9-6c8f-52ca-79ce-2aa6dca1a1de))

Reply:
This makes no sense. That open PR is for the Python 3.7 build. The Python 3.9 build has included libxcrypt in its host requirements for 2 months now

https://github.com/conda-forge/python-feedstock/blame/3.9/recipe/meta.yaml#L184

Reply:
I confirmed that libxcrypt is installed for the build and test sections for every Python version except 3.9

Reply:
I figured it out! For some reason the Python 3.9 build is installing the pypy version of python and not the cpython version

```
# 3.9 build
python:                     3.9.18-1_73_pypy            conda-forge

# 3.8 build
python:                   3.8.18-hd12c33a_1_cpython   conda-forge
```

That's pretty strange that only the 3.9 build is installing the pypy version

Reply:
While confusing, I don't think there is any problem with adding libxcrypt to the host requirements on linux. If a cython version of python is installed into the environment, libxcrypt will always be installed anyways